File: data/tutorial_wp/non-ghost-access-ghost.c
/* run.config
   EXIT: 1
   STDOPT:
*/

/*@
  requires n * 42 <= INT_MAX;    
*/
int sum_42(int n){
  int x = 0 ;
  //@ ghost int r = 42 ;
  /*@
    loop invariant x == i * r;
  */
  for(int i = 0; i < n; ++i){
    x += r;
  }
  return x;
}

File: transpiled_rust/tutorial_wp/non-ghost-access-ghost.rs
fn sum_42(n: i32) -> i32 {
    let mut x: i32 = 0;
    let r: i32 = 42;
    
    let mut i: i32 = 0;
    while i < n {
        x += r;
        i += 1;
    }
    x
}